compiler/rustc_infer/src/infer/equate.rs

```diff
@@ -110,6 +110,24 @@ impl<'tcx> TypeRelation<'tcx> for Equate<'_, '_, 'tcx> {
                         .obligations,
                 );
             }
+            (&ty::GeneratorWitness(a_types), &ty::GeneratorWitness(b_types)) => {
+                let a_types = infcx.tcx.anonymize_bound_vars(a_types);
+                let b_types = infcx.tcx.anonymize_bound_vars(b_types);
+                if a_types.bound_vars() == b_types.bound_vars() {
+                    let (a_types, b_types) = infcx.replace_bound_vars_with_placeholders(
+                        a_types.map_bound(|a_types| (a_types, b_types.skip_binder())),
+                    );
+                    for (a, b) in std::iter::zip(a_types, b_types) {
+                        self.relate(a, b)?;
+                    }
+                } else {
+                    self.fields.infcx.super_combine_tys(
+                        self,
+                        infcx.tcx.mk_generator_witness(a_types),
+                        infcx.tcx.mk_generator_witness(b_types),
+                    )?;
```

Comment:
i would expect this to error instead 🤔

Reply:
Yeah fair, I was going for a defensive approach of "fall back to old behavior", lol

Reply:
Ugh, so this causes things to fail in the case we have unevaluated consts, which after evaluating/normalization now have fewer regions than they had before (because the Unevaluated form may capture parent substs which mention things with regions). 

I think a fallback is needed, or else we need to do something like `normalize_erasing_regions` when constructing the generator interior.

Reply:
Could use advice @lcnr.

Reply:
Actually, just plain normalizing the generator interior types worked, no need to `normalize_erasing_regions`.